src/selector_gui.cpp

```diff
@@ -0,0 +1,375 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file selector_gui.cpp A composable selector widget. */
+
+#include "stdafx.h"
+#include "company_base.h"
+#include "company_gui.h"
+#include "strings_func.h"
+#include "table/sprites.h"
+#include "selector_gui.h"
+#include "widget_type.h"
+#include "widgets/selector_widget.h"
+#include "zoom_func.h"
+
+#include "safeguards.h"
+
+/**
+ * Create a new NWidgetBase with all the widgets needed by the SelectorWidget widget
+ * This function is made to be passed in NWidgetFunction, when creating UI for a parent window
+ * @see NWidgetFunction
+ */
+std::unique_ptr<NWidgetBase> SelectorWidget::MakeSelectorWidgetUI()
+{
+	static constexpr NWidgetPart widget_ui[] = {
+		NWidget(WWT_PANEL, COLOUR_BROWN),
+			NWidget(WWT_EDITBOX, COLOUR_BROWN, WID_SELECTOR_EDITBOX), SetFill(1, 0), SetResize(1, 0), SetPadding(2),
+					SetDataTip(STR_LIST_FILTER_TOOLTIP, STR_LIST_FILTER_TOOLTIP),
+		EndContainer(),
+		NWidget(NWID_VERTICAL),
+			NWidget(NWID_HORIZONTAL),
+				NWidget(WWT_MATRIX, COLOUR_BROWN, WID_SELECTOR_MATRIX),
+						SetScrollbar(WID_SELECTOR_SCROLLBAR), SetResize(1, 1), SetMatrixDataTip(1, 0, STR_NULL), SetFill(1, 1),
+				NWidget(NWID_VSCROLLBAR, COLOUR_BROWN, WID_SELECTOR_SCROLLBAR),
+			EndContainer(),
+			NWidget(NWID_HORIZONTAL, NC_EQUALSIZE),
+				NWidget(WWT_PUSHTXTBTN, COLOUR_BROWN, WID_SELECTOR_HIDEALL),
+						SetDataTip(STR_SELECTOR_WIDGET_DISABLE_ALL, STR_SELECTOR_WIDGET_TOOLTIP_DISABLE_ALL), SetResize(1, 0), SetFill(1, 0),
+				NWidget(WWT_PUSHTXTBTN, COLOUR_BROWN, WID_SELECTOR_SHOWALL),
+						SetDataTip(STR_SELECTOR_WIDGET_ENABLE_ALL, STR_SELECTOR_WIDGET_TOOLTIP_ENABLE_ALL), SetResize(1, 0), SetFill(1, 0),
```

Comment:
Selection buttons elsewhere are `All` before `None`, as suggested above and in #12847.

```suggestion
				NWidget(WWT_PUSHTXTBTN, COLOUR_BROWN, WID_SELECTOR_SHOWALL),
						SetDataTip(STR_SELECTOR_WIDGET_ENABLE_ALL, STR_SELECTOR_WIDGET_TOOLTIP_ENABLE_ALL), SetResize(1, 0), SetFill(1, 0),
				NWidget(WWT_PUSHTXTBTN, COLOUR_BROWN, WID_SELECTOR_HIDEALL),
						SetDataTip(STR_SELECTOR_WIDGET_DISABLE_ALL, STR_SELECTOR_WIDGET_TOOLTIP_DISABLE_ALL), SetResize(1, 0), SetFill(1, 0),
```